routing/routing_tests/routing_session_test.cpp

```diff
@@ -44,6 +44,36 @@ class DummyRouter : public IRouter
 static vector<m2::PointD> kTestRoute = {{0., 1.}, {0., 2.}, {0., 3.}, {0., 4.}};
 static auto kRouteBuildingMaxDuration = seconds(30);
 
+class TimedSignal
+{
+public:
+  TimedSignal() : m_lock(m_waitingMutex) {}
+
+  void Signal()
+  {
+    lock_guard<mutex> guard(m_waitingMutex);
+    m_flag = true;
+    cv.notify_one();
+  }
+
+  bool WaitUntil(steady_clock::time_point const & time)
+  {
+    cv.wait_until(m_lock, time, [this, &time]{return m_flag || steady_clock::now() > time;});
+    return m_flag;
+  }
+
+  void Reset()
+  {
+    m_flag = false;
+  }
+
+private:
+  mutex m_waitingMutex;
+  unique_lock<mutex> m_lock;
```

Comment:
Don't do that - move unique_lock m_lock to WaitUntil(). There're no need in a such field.